### PR TITLE
feat: add a canned elastic stage

### PR DESCRIPTION
**Pull request by rb-determined-ai**

### CODE REVIEW

File: devcluster/config.py [hunk 1]
```diff
@@ -180,13 +192,97 @@ def build_stage(
...
+                },
+            ),
+            "ElasticConfig.environment must be a dict of strings to strings",
+        )
+        self.data_dir = read_path(config.get("data_dir"))
+        self.container_name = str(config.get("container_name", "determined_elastic"))
+        self.name = str(config.get("name", "elastic"))
+        self.image_name = str(
+            config.get(
+                "image_name", "docker.elastic.co/elasticsearch/elasticsearch:7.9.3"
```
Comment (rb-determined-ai) [inline devcluster/config.py:243]: Why all this docker.elastic.co?  Why not just `elasticsearch:VERSION`?  And why does the [elasticsearch docker page](https://hub.docker.com/_/elasticsearch/) say that `7.9.3` isn't a supported image tag?

Comment (ioga) [reply devcluster/config.py:243]: >Why all this docker.elastic.co? Why not just elasticsearch:VERSION?

I was just porting this from `make -C tools`.

>And why does the elasticsearch docker page say that 7.9.3 isn't a supported image tag?

here it is, no? https://hub.docker.com/_/elasticsearch/?tab=tags&page=1&ordering=last_updated&name=7.9.3

File: devcluster/config.py [hunk 2]
```diff
@@ -69,7 +77,7 @@ class StageConfig(metaclass=abc.ABCMeta):
 
     @staticmethod
     def read(config: typing.Any, temp_dir: str) -> "StageConfig":
-        allowed = {"db", "master", "agent", "custom", "custom_docker"}
+        allowed = {"db", "elastic", "master", "agent", "custom", "custom_docker"}
```
Comment (stoksc) [inline devcluster/config.py:80]: idk how the ordering works in devcluster, but you may want elastic first. i imagine it's the thing you wanna bring down least?

Comment (rb-determined-ai) [reply devcluster/config.py:80]: Ok, I can adjust that.  Only the experiment config really matters but I generally like having things ordered consistently when it's not too much work.

File: devcluster/config.py (hunk 1 above)
Comment (rb-determined-ai) [reply devcluster/config.py:243]: > I was just porting this from make -C tools.

@stoksc ?

> here it is, no?

Yeah you can download 7.9.3, but that link clearly says the supported elasticsearch tags are  7.14.0 and 6.8.18.  (@stoksc should I upgrade this tag to 7.14.0?)

File: devcluster/config.py (hunk 2 above)
Comment (stoksc) [reply devcluster/config.py:80]: i just meant like, the state machine or devcluster. elastic is the slowest for sure.

Comment (rb-determined-ai) [reply devcluster/config.py:80]: lol, edited my comment

Comment (stoksc) [reply devcluster/config.py]: expconf nightmares

File: devcluster/config.py (hunk 1 above)
Comment (rb-determined-ai) [reply devcluster/config.py]: Looks like 7.14.0 seems to work fine.

**Review by ioga — APPROVED** [review]
plain `elasticsearch:7.14.0` seems to also be working fine

**Pull request merged**